RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitConverter.cc

```diff
@@ -120,7 +120,9 @@ namespace cms {
       DetId detIdObject(detid);
       const GeomDetUnit* genericDet = geom->idToDetUnit(detIdObject);
       auto gind = genericDet->index();
-      assert(gind < 2000);
+      // FIXME to be changed to support Phase2
+      if (gind >= int(gpuClustering::MaxNumModules))
+        continue;
```

Comment:
For my education, is this loop needed for the legacy cluster input use case?

Reply:
yes.

Reply:
Actually it is a bit more involved. This producer shall produce the very same products of its "SoAonCPU&2Legacy" buddy: SiPixelRecHitSoAFromLegacy. Most probably it could produce an empty product.
Or SiPixelRecHitSoAFromLegacy shall be split in two (or better everything merged in one, (actually fully refactorized)).
See my presentation to the DPG code-transfer workshop.